src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -231,12 +236,33 @@ headers_iterator(void *data, enum MHD_ValueKind kind, const char *key, const cha
 }
 
 static int
+get_static_file(const char *dir, const char *url)
+{
+    int ret;
+    char path[PATH_MAX];
+
+    /* url given by microhttpd starts from /. For instance,
+     * https://www.solettaproject.com => url == /
+     * https://www.solettaproject.com/thankyou => url == /thankyou
+     */
+    ret = snprintf(path, sizeof(path), "%s/%s", dir,
```

Comment:
good catch with ../../ 
About the other observation i will address in a different commit since it's already happening.